### ISSUE TITLE
Following steps to update libimobiledevice and ideviceinstaller don't seem to work

### ISSUE THREAD

**Issue by alardizabal** [flutter/flutter#10633]
I am seeing the below message in flutter doctor.

```
    ✗ libimobiledevice and ideviceinstaller are not installed or require updating. To update, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
```

Following the steps listed in the message don't seem to resolve the issue. Step-by-step shown below.

```

☆ flutter doctor
[✓] Flutter (on Mac OS X 10.12.5 16F73, locale en-US, channel master)
    • Flutter at /Users/albertlardizabal/dev/flutter
    • Framework revision 0f277fcc8a (6 days ago), 2017-06-06 17:19:27 +0200
    • Engine revision 8686a45810
    • Tools Dart version 1.24.0-dev.3.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.3)
    • Android SDK at /Users/albertlardizabal/Library/Android/sdk
    • Platform android-25, build-tools 25.0.3
    • Java binary at: /Applications/Android Studio 3.0 Preview.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b736)

[-] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    ✗ libimobiledevice and ideviceinstaller are not installed or require updating. To update, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.0
    • CocoaPods version 1.2.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Gradle version 3.2
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.0 Preview.app/Contents
    • Gradle version 4.0-milestone-1
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b736)

[✓] IntelliJ IDEA Community Edition (version 2017.1.4)
    • Flutter plugin version 14.0
    • Dart plugin version 171.4694.29

[✓] Connected devices
    • Pixel        • FA71A0302167                             • android-arm • Android 7.1.2 (API 25)
    •              • c0e109b7e55a826627aa0f3e79af8931534f3554 • ios         • iOS  ()
    • Posse iPhone • ac24c3de3cbe39efed542095769ce83fc89630f7 • ios         • iOS 10.3.1 (14E304)
    • iPhone 6     • 97A6FC25-AE09-4302-A8B3-565058389FE7     • ios         • iOS 10.3 (simulator)


☆ brew update
Already up-to-date.


☆ brew uninstall --ignore-dependencies libimobiledevice
Uninstalling /usr/local/Cellar/libimobiledevice/HEAD-53fede0_2... (66 files, 993.6KB)


☆ brew install --HEAD libimobiledevice
==> Using the sandbox
==> Cloning https://git.libimobiledevice.org/libimobiledevice.git
Updating /Users/albertlardizabal/Library/Caches/Homebrew/libimobiledevice--git
==> Checking out branch master
==> ./autogen.sh
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/libimobiledevice/HEAD-53fede0_2 --without-cython
==> make install
🍺  /usr/local/Cellar/libimobiledevice/HEAD-53fede0_2: 66 files, 993.6KB, built in 3 minutes 4 seconds


☆ brew install ideviceinstaller
Warning: ideviceinstaller 1.1.0_2 is already installed


☆ flutter doctor
[✓] Flutter (on Mac OS X 10.12.5 16F73, locale en-US, channel master)
    • Flutter at /Users/albertlardizabal/dev/flutter
    • Framework revision 0f277fcc8a (6 days ago), 2017-06-06 17:19:27 +0200
    • Engine revision 8686a45810
    • Tools Dart version 1.24.0-dev.3.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.3)
    • Android SDK at /Users/albertlardizabal/Library/Android/sdk
    • Platform android-25, build-tools 25.0.3
    • Java binary at: /Applications/Android Studio 3.0 Preview.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b736)

[-] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    ✗ libimobiledevice and ideviceinstaller are not installed or require updating. To update, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.0
    • CocoaPods version 1.2.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Gradle version 3.2
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] Android Studio
    • Android Studio at /Applications/Android Studio 3.0 Preview.app/Contents
    • Gradle version 4.0-milestone-1
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b736)

[✓] IntelliJ IDEA Community Edition (version 2017.1.4)
    • Flutter plugin version 14.0
    • Dart plugin version 171.4694.29

[✓] Connected devices
    • Pixel        • FA71A0302167                             • android-arm • Android 7.1.2 (API 25)
    •              • c0e109b7e55a826627aa0f3e79af8931534f3554 • ios         • iOS  ()
    • Posse iPhone • ac24c3de3cbe39efed542095769ce83fc89630f7 • ios         • iOS 10.3.1 (14E304)
    • iPhone 6     • 97A6FC25-AE09-4302-A8B3-565058389FE7     • ios         • iOS 10.3 (simulator)
```

**Comment by eseidelGoogle** [comment]
Looks like you have to `brew uninstall libimobiledevice` first.  FYI @cbracken

**Comment by alardizabal** [comment]
I've seen this several times and I have tried to uninstall first. It does uninstall and installation seems successful but then 'flutter doctor' will still show the same message (not installed or needs updating). I'll make sure to screenshot that if I see it again in the future.

**Comment by cbracken** [comment]
Out of curiosity -- what kind of device is `c0e109b7e55a826627aa0f3e79af8931534f3554` -- device type and OS?

**Comment by alardizabal** [comment]
Ah, I forget I have so many devices plugged in. That's my iPhone 6 running the iOS 11 beta.  I haven't even gotten around to filing that bug (device details don't display). I just noticed that the libimobiledevice warning only displays when it is plugged in.

**Comment by cbracken** [comment]
Thanks for confirming. I'm also able to repro with an iOS 11 beta device. I haven't confirmed yet whether we'll be dependent on a libimobiledevice update for iOS 11 support, but should know in relatively short order.

FYI there's at least one other blocker for iOS 11 at the moment: #10602. Apps themselves run fine on iOS 11, but hot-reload development is blocked by that issue.

**Comment by cbracken** [comment]
As of #10759, iOS 11 devices should now appear correctly in `flutter devices` and doctor output.

The `libimobiledevice` warning still exists for the moment, but once I've eliminated a couple more dependencies on `libimobiledevice`, it'll be safe to remove it.

**Comment by cbracken** [comment]
This issue should now be resolved. Following the steps in `flutter doctor` output (one last time) should be sufficient.

**Comment by gabrieloshiro** [comment]
Sorry to reopen this issue, but I am following the official tutorial today (Nov 16, 2017) and I have the exact same issue

**Comment by cbracken** [comment]
@gabrieloshiro thanks for reporting! Would you mind pasting the output of a `flutter doctor` run on your machine?

**Comment by smzranz** [comment]
[✓] Flutter (Channel beta, v0.2.8, on Mac OS X 10.12.6 16G29, locale en-IN)
    • Flutter version 0.2.8 at /Users/shamshir/Documents/Flutter.io/newFlutter/flutter
    • Framework revision b397406561 (3 weeks ago), 2018-04-02 13:53:20 -0700
    • Engine revision c903c217a1
    • Dart version 2.0.0-dev.43.0.flutter-52afcba357

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/shamshir/Documents/AndroidSDk/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = /Users/shamshir/Documents/AndroidSDk/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 9.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.1, Build version 9B55
    ✗ ideviceinstaller is not installed; this is used to discover connected iOS devices.
      To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install:
        brew install ios-deploy
    • CocoaPods version 1.3.1

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] VS Code (version 1.22.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension version 2.11.2

[✓] Connected devices (1 available)
    • iPhone 8 Plus • BF3E917E-C2F6-4DA7-9A10-4015C32447F4 • ios • iOS 11.1 (simulator)

**Comment by rakz25** [comment]
Facing the same issue and following is the output when run brew install on terminal 

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v0.5.8-pre.187, on Mac OS X 10.12.6 16G29, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK 28.0.1)
[!] iOS toolchain - develop for iOS devices (Xcode 9.1)
    ✗ libimobiledevice and ideviceinstaller are not installed. To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ! CocoaPods out of date (1.5.0 is recommended).
        CocoaPods is used to retrieve the iOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without resolving iOS dependencies with CocoaPods, plugins will not work on iOS.
        For more info, see https://flutter.io/platform-plugins
      To upgrade:
        brew upgrade cocoapods
        pod setup
[✓] Android Studio (version 3.1)
[✓] Connected devices (1 available)

! Doctor found issues in 1 category.
Rakeshs-MacBook-Air:experiment_01 rakesh$ brew install
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (dart-lang/dart, homebrew/core).
==> New Formulae
gradio                                                      prettier                                                    thors-serializer
==> Updated Formulae
adr-tools                convox                   fonttools                ipython                  mitmproxy                pgplot                   tippecanoe
angular-cli              dart-lang/dart/dart@2    fruit                    ipython@5                mkcert                   pgpool-ii                tmux-xpanes
annie                    dep                      fwup                     isl                      modd                     php-code-sniffer         tomcat@8
ansible                  distcc                   gcc                      jenkins                  monero                   phplint                  ttfautohint
arangodb                 docfx                    gdb                      jenkins-job-builder      mongo-cxx-driver         picard-tools             tup
ark                      duply                    gdbm                     jlog                     mr                       plantuml                 typescript
asdf                     dxpy                     geoserver                joplin                   mrtg                     pmd                      uhd
azure-cli                dynare                   git-lfs                  jpeg-turbo               mysql                    pony-stable              urh
b2-tools                 eigen                    gitbucket                json-fortran             mysql-client             ponyc                    utf8proc
ballerina                elixir                   gnome-builder            juju                     mysql@5.5                pspg                     vault
bartycrouch              emscripten               gnome-latex              kompose                  mysql@5.6                qbs                      vips
bit                      erlang                   go-jira                  kops                     mysql@5.7                quicktype                vnu
bitcoin                  erlang@20                goenv                    libdazzle                nativefier               root                     wabt
bitrise                  etcd                     gomplate                 libmagic                 nginx                    sbcl                     webpack
bitwarden-cli            ethereum                 goreleaser               libpng                   nsd                      sbt                      xonsh
blockhash                exercism                 grafana                  librealsense             numpy                    shellharden              yarn
buildifier               f3                       groonga                  link-grammar             nuxeo                    skinny                   youtube-dl
cabextract               faas-cli                 gst-editing-services     lxc                      openapi-generator        sratoolkit               zanata-client
ccrypt                   fabric                   gupnp                    mapcrafter               openblas                 swift-protobuf
ceres-solver             file-formula             haproxy                  mbedtls                  opencoarrays             swiftformat
chrome-cli               firebase-cli             hive                     megatools                paket                    syncthing
citus                    fluent-bit               hopenpgp-tools           mikutter                 pango                    teleport
conan                    fn                       hugo                     mill                     pazpar2                  terragrunt
consul                   folly                    ipfs                     minimal-racket           pgcli                    tig

brew install [--debug] [--env=(std|super)] [--ignore-dependencies|--only-dependencies] [--cc=compiler] [--build-from-source|--force-bottle] [--include-test] [--devel|--HEAD] [--keep-tmp] [--build-bottle] [--force] [--verbose] [--display-times] formula [options ...]:
    Install formula.

    formula is usually the name of the formula to install, but it can be specified
    in several different ways. See [SPECIFYING FORMULAE](#specifying-formulae).

    If --debug (or -d) is passed and brewing fails, open an interactive debugging
    session with access to IRB or a shell inside the temporary build directory.

    If --env=std is passed, use the standard build environment instead of superenv.

    If --env=super is passed, use superenv even if the formula specifies the
    standard build environment.

    If --ignore-dependencies is passed, skip installing any dependencies of
    any kind. If they are not already present, the formula will probably fail
    to install.

    If --only-dependencies is passed, install the dependencies with specified
    options but do not install the specified formula.

    If --cc=compiler is passed, attempt to compile using compiler.
    compiler should be the name of the compiler's executable, for instance
    gcc-8 for gcc 8, gcc-4.2 for Apple's GCC 4.2, or gcc-4.9 for a
    Homebrew-provided GCC 4.9. In order to use LLVM's clang, use
    llvm_clang. To specify the Apple-provided clang, use clang. This
    parameter will only accept compilers that are provided by Homebrew or
    bundled with macOS. Please do not file issues if you encounter errors
    while using this flag.

    If --build-from-source (or -s) is passed, compile the specified formula from
    source even if a bottle is provided. Dependencies will still be installed
    from bottles if they are available.

    If HOMEBREW_BUILD_FROM_SOURCE is set, regardless of whether --build-from-source was
    passed, then both formula and the dependencies installed as part of this process
    are built from source even if bottles are available.

    If --force-bottle is passed, install from a bottle if it exists for the
    current or newest version of macOS, even if it would not normally be used
    for installation.

    If --include-test is passed, install testing dependencies. These are only
    needed by formulae maintainers to run brew test.

    If --devel is passed, and formula defines it, install the development version.

    If --HEAD is passed, and formula defines it, install the HEAD version,
    aka master, trunk, unstable.

    If --keep-tmp is passed, the temporary files created during installation
    are not deleted.

    If --build-bottle is passed, prepare the formula for eventual bottling
    during installation.

    If --force (or -f) is passed, install without checking for previously
    installed keg-only or non-migrated versions

    If --verbose (or -v) is passed, print the verification and postinstall steps.

    If --display-times is passed, install times for each formula are printed
    at the end of the run.

    Installation options specific to formula may be appended to the command,
    and can be listed with brew options formula.

brew install --interactive [--git] formula:
    If --interactive (or -i) is passed, download and patch formula, then
    open a shell. This allows the user to run ./configure --help and
    otherwise determine how to turn the software package into a Homebrew
    formula.

    If --git (or -g) is passed, Homebrew will create a Git repository, useful for
    creating patches to the software.

Error: Invalid usage: This command requires a formula argument

**Comment by zoechi** [comment]
@rakz25 please create a new issue with the issue template filled.
This issue is long closed and it's likely that your issue is different and the right people won't see your comments on closed issues.

**Comment by nisimjoseph** [comment]
I am getting this error, see the console output:
`
==> Cloning https://git.libimobiledevice.org/libimobiledevice.git
Updating /Users/nisim.j/Library/Caches/Homebrew/libimobiledevice--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 0584aa9 idevicepair: Fix udid being freed before printing error message containing udid
==> ./autogen.sh
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/libimobiledevice/HEAD-0584aa9_3 --without-cython --enable-debug-code
==> make install
Last 15 lines from /Users/nisim.j/Library/Logs/Homebrew/libimobiledevice/03.make:
        int res = pthread_create(thread, NULL, thread_func, data);
                                         ^
thread.c:52:23: error: use of undeclared identifier 'NULL'
        pthread_join(thread, NULL);
                             ^
thread.c:61:28: error: use of undeclared identifier 'NULL'
        pthread_mutex_init(mutex, NULL);
                                  ^
7 errors generated.
make[1]: *** [thread.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../include -I.. -Wall -Wextra -Wmissing-declarations -Wredundant-decls -Wshadow -Wpointer-arith -Wwrite-strings -Wswitch-default -Wno-unused-parameter -fsigned-char -fvisibility=hidden -I/usr/local/Cellar/libplist/2.0.0_1/include -I/usr/local/Cellar/usbmuxd/HEAD-c75605d_1/include -I/usr/local/Cellar/libplist/2.0.0_1/include -I/usr/local/Cellar/openssl/1.0.2q/include -D_LARGE_FILES=1 -g -O2 -c debug.c -o debug.o >/dev/null 2>&1
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../include -I.. -Wall -Wextra -Wmissing-declarations -Wredundant-decls -Wshadow -Wpointer-arith -Wwrite-strings -Wswitch-default -Wno-unused-parameter -fsigned-char -fvisibility=hidden -I/usr/local/Cellar/libplist/2.0.0_1/include -I/usr/local/Cellar/usbmuxd/HEAD-c75605d_1/include -I/usr/local/Cellar/libplist/2.0.0_1/include -I/usr/local/Cellar/openssl/1.0.2q/include -D_LARGE_FILES=1 -g -O2 -c userpref.c -o userpref.o >/dev/null 2>&1
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../include -I.. -Wall -Wextra -Wmissing-declarations -Wredundant-decls -Wshadow -Wpointer-arith -Wwrite-strings -Wswitch-default -Wno-unused-parameter -fsigned-char -fvisibility=hidden -I/usr/local/Cellar/libplist/2.0.0_1/include -I/usr/local/Cellar/usbmuxd/HEAD-c75605d_1/include -I/usr/local/Cellar/libplist/2.0.0_1/include -I/usr/local/Cellar/openssl/1.0.2q/include -D_LARGE_FILES=1 -g -O2 -c utils.c -o utils.o >/dev/null 2>&1
make: *** [install-recursive] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting
`

**Comment by ergunkocak** [comment]
Did anyone has a solution?

**Comment by nisimjoseph** [comment]
I couldn't solve it until now. the only solution I found is to install the current `brew install libimobiledevice` without --HEAD. this is working for iOS simulator and Android.
to test it in real iOS device I run the Flutter output directly in XCode and it runs.
waiting for the fix of "libimobiledevice" package to have it work better.

**Comment by nisimjoseph** [comment]
I FOUND IT!!!

The is not the fault of `libimobiledevice` package. The issue exists in a different place. I get the repository and compiled it so I found the issue inside Mac OSX `pthread.h` file located at `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/pthread`.
That file has a missing include line: `#include <stddef.h>`. I think it because of latest updates of the OS
To add it do the following steps:
1. open the folder location in the Finder, you can do it in command line: `open /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/pthread`.
2. Find the file `pthread.h` and CMD+I (info)
3. At the bottom of it (Permission section) click "+" and add your user name and give yourself `read & write` permissions. not we can add that missing line.
4. Find the code `#include <_types.h>` and add the line `#include <stddef.h>` after it.
5. Save the file.
6. Run `brew uninstall --ignore-dependencies libimobiledevice`
7. Run `brew install --HEAD libimobiledevice`

**Now all will compile and you will get all green:**
![image](https://user-images.githubusercontent.com/2305992/54817592-01315780-4ca0-11e9-925d-977dc2182ac9.png)

**Comment by JCCP3** [comment]
> I FOUND IT!!!
> 
> The is not the fault of `libimobiledevice` package. The issue exists in a different place. I get the repository and compiled it so I found the issue inside Mac OSX `pthread.h` file located at `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/pthread`.
> That file has a missing include line: `#include <stddef.h>`. I think it because of latest updates of the OS
> To add it do the following steps:
> 
> 1. open the folder location in the Finder, you can do it in command line: `open /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/pthread`.
> 2. Find the file `pthread.h` and CMD+I (info)
> 3. At the bottom of it (Permission section) click "+" and add your user name and give yourself `read & write` permissions. not we can add that missing line.
> 4. Find the code `#include <_types.h>` and add the line `#include <stddef.h>` after it.
>    ![image](https://user-images.githubusercontent.com/2305992/54817777-65541b80-4ca0-11e9-8cbf-eb19b5ed9280.png)
> 5. Save the file.
> 6. Run `brew uninstall --ignore-dependencies libimobiledevice`
> 7. Run `brew install --HEAD libimobiledevice`
> 
> **Now all will compile and you will get all green:**
> ![image](https://user-images.githubusercontent.com/2305992/54817592-01315780-4ca0-11e9-925d-977dc2182ac9.png)

thank you this solved my problem

**Comment by dhavalagile** [comment]
Hi, I will try all possible ways suggested above but unable to solve usbmuxd and libimobiledevice issues.
![Screenshot 2019-04-02 at 6 13 29 PM](https://user-images.githubusercontent.com/25456773/55403285-1d23db80-5573-11e9-9732-798da2b5c788.png)

**Comment by nisimjoseph** [comment]
@dhavalagile did you get errors when running `brew install --HEAD libimobiledevice` ?
send here the console output.

**Comment by dhavalagile** [comment]
I am getting this error. Please check the console output screenshot.
![Screenshot 2019-04-03 at 10 33 58 AM](https://user-images.githubusercontent.com/25456773/55454077-1f7e4800-55fc-11e9-924c-b891fa97f707.png)

**Comment by nisimjoseph** [comment]
@dhavalagile you need to `brew uninstall usbmuxd` and reinstall it using the command `brew install --HEAD usbmuxd`,
the log complain it has older version and not the HEAD one.

**Comment by dhavalagile** [comment]
@nisimjoseph I have try brew uninstall usbmuxd but gives me an error. and suggest me use brew uninstall --ignore-dependencies usbmuxd.

But when I am trying brew install --HEAD usbmuxd it said remote branch master not found.
![Screenshot 2019-04-03 at 4 44 12 PM](https://user-images.githubusercontent.com/25456773/55474802-c6300c00-562f-11e9-8319-e98c93987a3b.png)

**Comment by nisimjoseph** [comment]
@dhavalagile yes, run `brew uninstall --ignore-dependencies usbmuxd`
About the error, I don't know, but uninstall and run install again.

**Comment by NirubanThiyagarajan** [comment]
> Did anyone has a solution?

Try brew link usbmuxd. This solved it for me.

**Comment by testerSunshine** [comment]
Cloning https://git.libimobiledevice.org/libimobiledevice.git 
Can not access
please update！！！

**Comment by nisimjoseph** [comment]
@testerSunshine https://github.com/libimobiledevice/libimobiledevice

**Comment by zackdorward** [comment]
This didn't work for me. Here is my flutter doctor output: 
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.5.4, on Mac OS X 10.14.4 18E227, locale en-CA)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[!] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
    ✗ libimobiledevice and ideviceinstaller are not installed. To install with Brew, run:
        brew update
        brew install --HEAD usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
[✓] Android Studio (version 3.3)
[✓] Connected device (1 available)

! Doctor found issues in 1 category.

Note: Right now, the only connected device that my mac recognizes is my simulator, even though my iPhone 8 is currently plugged in. 

Here is my output when I run brew install --HEAD libimobiledevice:

Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ dir_s_mkdir - /Users/*******/Library/Logs/Homebrew/libimobiledevice

**Comment by nisimjoseph** [comment]
@zackdorward can you clear your log, uninstall the brew component above. and run the install again all list of brew install. send the clean log here. Hope there is more thing we can get help with.
The `dir_s_mkdir` does not exist on my machine as well and it not needed as I install all and it didn't prevent me from complete the install.

**Comment by zackdorward** [comment]
I'm assuming by the "brew component above" you want me to run **brew uninstall usbmuxd?**

**Comment by zackdorward** [comment]
I'm assuming by the "brew component above" you want me to run _brew uninstall usbmuxd_?

**Comment by zackdorward** [comment]
I'm assuming by the "brew component above" you want me to run 

> brew uninstall usbmuxd

?

**Comment by zackdorward** [comment]
Not sure what you mean by 

> the brew component above

if you want me to run  ### brew uninstall usbmuxd then here is the output

**Comment by zackdorward** [comment]
@nisimjoseph Not sure what you mean by "the brew component above"

But if you meant for me to run **brew uninstall usbmuxd** then here is the output: 

**Error: Refusing to uninstall /usr/local/Cellar/usbmuxd/HEAD-873252d_1
because it is required by ideviceinstaller, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies usbmuxd**

When I run **brew uninstall --ignore-dependencies usbmuxd** it uninstalls properly. 

Then I ran: 
**brew update
brew uninstall --ignore-dependencies libimobiledevice
brew install --HEAD libimobiledevice
brew install ideviceinstaller**

and here was the output: 

Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Formulae
plantuml                                 pmd
iMac:~ *******$         brew uninstall --ignore-dependencies libimobiledevice
Error: No such keg: /usr/local/Cellar/libimobiledevice
iMac:~ *******$         brew install --HEAD libimobiledevice
==> Installing dependencies for libimobiledevice: usbmuxd
==> Installing libimobiledevice dependency: usbmuxd
==> Downloading https://homebrew.bintray.com/bottles/usbmuxd-1.0.10_1.mojave.bot
######################################################################## 100.0%
==> Pouring usbmuxd-1.0.10_1.mojave.bottle.tar.gz
🍺  /usr/local/Cellar/usbmuxd/1.0.10_1: 13 files, 120KB
==> Installing libimobiledevice --HEAD
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ dir_s_mkdir - /Users/*******/Library/Logs/Homebrew/libimobiledevice
iMac:~ *******$         brew install ideviceinstaller
Warning: ideviceinstaller 1.1.0_4 is already installed and up-to-date
To reinstall 1.1.0_4, run `brew reinstall ideviceinstaller

**Comment by nisimjoseph** [comment]
@zackdorward Please install in the following sequence, including `brew install --HEAD usbmuxd` before installing `brew install --HEAD libimobiledevice`:

brew update
brew install --HEAD usbmuxd
brew link usbmuxd
brew install --HEAD libimobiledevice
brew install ideviceinstaller

**Comment by nisimjoseph** [comment]
Also, try to run `brew cleanup` before installing all the list.

**Comment by zackdorward** [comment]
@nisimjoseph 

iMac:~ Dorward$ brew update
Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Formulae
diff-pdf                   node                       swig
docker-credential-helper   pdftoipe                   unrar
fonttools                  poppler                    vulkan-headers
joplin                     sbcl
mosquitto                  step
iMac:~ Dorward$ brew install --HEAD usbmuxd
Error: usbmuxd 1.0.10_1 is already installed
To install HEAD_1, first run `brew unlink usbmuxd`.
Warning: Skipping (old) /usr/local/Cellar/usbmuxd/1.0.10_1 due to it being linked
iMac:~ Dorward$ brew link usbmuxd
Warning: Already linked: /usr/local/Cellar/usbmuxd/1.0.10_1
To relink: brew unlink usbmuxd && brew link usbmuxd
iMac:~ Dorward$ brew install --HEAD libimobiledevice
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory @ dir_s_mkdir - /Users/Dorward/Library/Logs/Homebrew/libimobiledevice
iMac:~ Dorward$ brew install ideviceinstaller
Warning: ideviceinstaller 1.1.0_4 is already installed and up-to-date
To reinstall 1.1.0_4, run `brew reinstall ideviceinstaller`

**Comment by nisimjoseph** [comment]
@zackdorward Ok, it a new issue.
so let's do something more hardcore to find the issue.
go to `https://github.com/libimobiledevice/libimobiledevice`, download the repository, and run the installation using CLI. all instruction is there.
the build will notify you of the real issue in the build. this is how I found the last issue.

**Comment by nisimjoseph** [comment]
@zackdorward Ok, it a new issue.
so let's do something more hardcore to find the issue.
go to `https://github.com/libimobiledevice/libimobiledevice/tree/0584aa90c93ff6ce46927b8d67887cb987ab9545`, download the repository *from this version*, and run the installation using CLI. all instruction is there.
the build will notify you of the real issue in the build. this is how I found the last issue.

I see a commit that probably caused it, the previous one supposes to work.

**Comment by zackdorward** [comment]
@nisimjoseph 

Last login: Tue Apr 30 23:58:27 on ttys000
iMac:~ Dorward$ ./autogen.sh
-bash: ./autogen.sh: No such file or directory
iMac:~ Dorward$ make
make: *** No targets specified and no makefile found.  Stop.
iMac:~ Dorward$ ./autogen.sh
-bash: ./autogen.sh: No such file or directory
iMac:~ Dorward$ make
make: *** No targets specified and no makefile found.  Stop.
iMac:~ Dorward$ cd /Users/Dorward/Downloads/libimobiledevice-0584aa90c93ff6ce46927b8d67887cb987ab9545 
iMac:libimobiledevice-0584aa90c93ff6ce46927b8d67887cb987ab9545 Dorward$ ./autogen.sh
glibtoolize: putting auxiliary files in '.'.
glibtoolize: linking file './ltmain.sh'
glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
glibtoolize: linking file 'm4/libtool.m4'
glibtoolize: linking file 'm4/ltoptions.m4'
glibtoolize: linking file 'm4/ltsugar.m4'
glibtoolize: linking file 'm4/ltversion.m4'
glibtoolize: linking file 'm4/lt~obsolete.m4'
configure.ac:31: installing './compile'
configure.ac:34: installing './config.guess'
configure.ac:34: installing './config.sub'
configure.ac:6: installing './install-sh'
configure.ac:6: installing './missing'
common/Makefile.am: installing './depcomp'
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... gcc3
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking build system type... x86_64-apple-darwin18.5.0
checking host system type... x86_64-apple-darwin18.5.0
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking how to convert x86_64-apple-darwin18.5.0 file names to x86_64-apple-darwin18.5.0 format... func_convert_file_noop
checking how to convert x86_64-apple-darwin18.5.0 file names to toolchain format... func_convert_file_noop
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin18.5.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fno-common -DPIC
checking if g++ PIC flag -fno-common -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin18.5.0 dyld
checking how to hardcode library paths into programs... immediate
checking for pkg-config... /usr/local/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libusbmuxd >= 1.1.0... yes
checking for libplist >= 1.11... yes
checking for libplist++ >= 1.11... yes
checking for ANSI C header files... (cached) yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking gcrypt.h usability... no
checking gcrypt.h presence... no
checking for gcrypt.h... no
checking for an ANSI C-conforming const... yes
checking for size_t... yes
checking for ssize_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint8_t... yes
checking for asprintf... yes
checking for strcasecmp... yes
checking for strdup... yes
checking for strerror... yes
checking for strndup... yes
checking for stpcpy... yes
checking for vasprintf... yes
checking endian.h usability... no
checking endian.h presence... no
checking for endian.h... no
checking whether byte ordering is bigendian... no
checking whether we need platform-specific build settings... no
checking for struct dirent.d_type... yes
checking for a Python interpreter with version >= 2.3... python
checking for python... /usr/bin/python
checking for python version... 2.7
checking for python platform... darwin
checking for python script directory... ${prefix}/lib/python2.7/site-packages
checking for python extension module directory... ${exec_prefix}/lib/python2.7/site-packages
checking for cython... no
configure: WARNING: cannot find 'cython' program. You should look at http://www.cython.org or install your distribution specific cython package.
checking for python2.7... (cached) /usr/bin/python
checking for a version of Python >= '2.1.0'... yes
checking for the distutils Python package... yes
checking for Python include path... /usr/bin/python-config is /usr/bin/python-config
-iwithsysroot /System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 -iwithsysroot /System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7
checking for Python library path... /usr/bin/python-config is /usr/bin/python-config
-L/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config -lpython2.7 -ldl -framework CoreFoundation
checking for Python site-packages path... /Library/Python/2.7/site-packages
checking python extra libraries... ('', '-ldl  -framework CoreFoundation')
checking python extra linking flags... -u _PyMac_Error /System/Library/Frameworks/Python.framework/Versions/2.7/Python
checking consistency of all components of python development environment... yes
checking for openssl >= 0.9.8... no
configure: error: OpenSSL support explicitly requested but OpenSSL could not be found
iMac:libimobiledevice-0584aa90c93ff6ce46927b8d67887cb987ab9545 Dorward$ make
make: *** No targets specified and no makefile found.  Stop.
iMac:libimobiledevice-0584aa90c93ff6ce46927b8d67887cb987ab9545 Dorward$ sudo make install
Password:
make: *** No rule to make target `install'.  Stop.
iMac:libimobiledevice-0584aa90c93ff6ce46927b8d67887cb987ab9545 Dorward$

**Comment by ShenYj** [comment]
> @dhavalagile yes, run `brew uninstall --ignore-dependencies usbmuxd`
> About the error, I don't know, but uninstall and run install again.

reinstall usbmuxd is not to work, I get the message:

[Shen@localhost ~/Desktop]$brew uninstall usbmuxd
Error: Refusing to uninstall /usr/local/Cellar/usbmuxd/1.0.10_1
because it is required by ios-webkit-debug-proxy, which is currently installed.
You can override this and force removal with:
  brew uninstall --ignore-dependencies usbmuxd
[Shen@localhost ~/Desktop]$brew uninstall usbmuxd --ignore-dependencies usbmuxd
Uninstalling /usr/local/Cellar/usbmuxd/1.0.10_1... (13 files, 120KB)
[Shen@localhost ~/Desktop]$brew install --HEAD usbmuxd
==> Cloning https://git.sukimashita.com/libusbmuxd.git
Updating /Users/Shen/Library/Caches/Homebrew/usbmuxd--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 6539b02 iproxy: Fix (newly introduced) timeout errors not being handled
==> ./autogen.sh
Last 15 lines from /Users/Shen/Library/Logs/Homebrew/usbmuxd/01.autogen.sh:
checking dynamic linker characteristics... darwin18.6.0 dyld
checking how to hardcode library paths into programs... immediate
checking for pkg-config... no
checking for libplist >= 1.11... no
configure: error: in `/private/tmp/usbmuxd-20190730-63119-1jvelyv':
configure: error: The pkg-config script could not be found or is too old.  Make sure it
is in your PATH or set the PKG_CONFIG environment variable to the full
path to pkg-config.

Alternatively, you may set the environment variables libplist_CFLAGS
and libplist_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

To get pkg-config, see <http://pkg-config.freedesktop.org/>.
See `config.log' for more details

READ THIS: https://docs.brew.sh/Troubleshooting
